app/src/app/components/specmath-modal/modal.component.ts

```diff
@@ -76,24 +97,94 @@ export class ModalComponent {
     specFiles: [],
     valid: false,
   };
+  resultSpec: File;
+  mergeConflicts: MergeConflict[];
+  loadingOperation = false;
 
-  constructor(readonly dialogRef: MatDialogRef<ModalComponent>) {
+  constructor(readonly dialogRef: MatDialogRef<ModalComponent>,
+              private cdr: ChangeDetectorRef,
+              private mockService: SpecMathService) {
     dialogRef.disableClose = true;
   }
 
-  nextStep(stepper: MatStepper): void {
-    if (this.currentStep === Steps.confirmOperation) {
-      this.dialogRef.close();
+  async nextStep(stepper: MatStepper) {
+    const currStep = stepList[this.currentStep];
+
+    if (currStep.lastBaseStep) {
+      this.loadingOperation = true;
+      await this.mergeOperation();
+      this.loadingOperation = false;
+      this.cdr.detectChanges();
+
+      if (!this.hasMergeConflicts) {
+        this.finalizeSteps();
+        return;
+      }
+    }
+
+    if (this.mergeConflictsResolved) {
+      if (!this.hasMergeConflicts) {
+        this.finalizeSteps();
+        return;
+      }
+
+      this.loadingOperation = true;
+      await this.sendResolvedConflicts();
+      this.loadingOperation = false;
+      this.finalizeSteps();
       return;
     }
 
-    this.currentStep = stepOptions[this.currentStep].nextStep;
-    stepper.selectedIndex = this.currentStep;
+    stepper.selectedIndex = ++this.currentStep;
+  }
+
+  previousStep(stepper: MatStepper) {
+    stepper.selectedIndex = --this.currentStep;
   }
 
-  previousStep(stepper: MatStepper): void {
-    this.currentStep = stepOptions[this.currentStep].previousStep;
-    stepper.selectedIndex = this.currentStep;
+  finalizeSteps() {
+    // ?This closing call will carry all the files to the parent component
+    this.dialogRef.close();
+  }
+
+  async mergeOperation() {
+    const mergeSet = await this.generateMergeSet();
+    const callResponse = await this.mockService.mergeSpecs(mergeSet).toPromise();
+
+    switch (callResponse?.status) {
+      case 'conflicts':
+        this.mergeConflicts = callResponse.conflicts;
+        break;
+      case 'success':
+        this.resultSpec = new File([callResponse.result], this.specNameInputOptions.newFileName);
```

Comment:
nit: It's probably good practice to have a `break` after the last case so it's not missed during future updates.

Reply:
Updated